qiskit/pulse/macros.py

```diff
@@ -104,3 +112,173 @@ def measure_all(backend) -> Schedule:
+            waveform to 1. The default is ``True`` and the amplitude is constrained to 1.
+        chunk_size: Size of a single chunk in units of dt. This must satisfy the
+            device timing constraint for the pulse granularity.
+        min_chunk_number: Minimum chunk number to enable the chunk division.
+            This macro plays a single :class:`.GaussianSquare` pulse when
+            the pulse duration is sufficiently short.
+
+    Returns:
+        ScheduleBlock of the chunked GaussianSquare pulse sequence.
+
+    Raises:
+        PulseError: When duration is parameterized.
+    """
+    # TODO this must be called from proper pulse compiler once implemented.
+    #  i.e. Standard GaussianSquare is implicitly replaced with this based on a compiler option.
+    #  Experimentalist doesn't need to take care of chunking like this.
+
+    if not isinstance(duration, int):
+        raise PulseError(
+            "Parameterized duration cannot be applied to the chunked pulse. "
+            "Specify an integer duration in units of dt."
+        )
+
+    if chunk_size % granularity != 0:
+        _new_size = granularity * int(chunk_size / granularity)
+        warnings.warn(
+            f"Chunk size of {chunk_size} dt doesn't match with the device timing constraint. "
+            f"The value is rounded to the nearest valid size of {_new_size} dt.",
+            UserWarning,
+        )
+        chunk_size = _new_size
+    width = duration - 2 * sigma * risefall_sigma_ratio
+    n_chunks = int(width / chunk_size)
+
+    schedule = ScheduleBlock(name=name, alignment_context=AlignSequential())
+
+    if n_chunks < min_chunk_number:
+        # Pulse is very short. No need to apply chunk division.
+        # Usually chunk division increases payload size because rising and falling edges
+        # are submitted down to the device in the waveform format.
+        # Switch back to the standard GaussianSquare pulse to compress payload.
+        valid_duration = granularity * int(duration / granularity)
+        schedule.append(
+            instructions.Play(
+                pulse.GaussianSquare(
+                    duration=valid_duration,
+                    amp=amp,
+                    sigma=sigma,
+                    risefall_sigma_ratio=risefall_sigma_ratio,
+                    name=name,
+                    limit_amplitude=limit_amplitude,
+                ),
+                channel=channel,
+            ),
+            inplace=True,
+        )
+    else:
+        # Extra flat-top duration absorbed by ramps, i.e. waveform
+        chunked_flat_top_size = n_chunks * chunk_size
+        total_slack = width - chunked_flat_top_size
+
+        # Effective pulse edge duration
+        # This must guarantee the duration > sigma * risefall_sigma_ratio
+        # It rounds up duration rather than truncating.
+        edge_size = granularity * math.ceil(
+            (sigma * risefall_sigma_ratio + total_slack / 2) / granularity
+        )
```

Comment:
Have you considered making `edge_size = granularity * math.ceil(sigma * risefall_sigma_ratio / granularity)` and putting the slack into one of the chunks (so one chunk is slightly larger than the rest)? As you mention that this would commonly be used in a scan, it might be more efficient to keep the rise/fall pulses identical across circuits and have just some of the constant chunks be varied rather than having varied flat sections in different rise/fall pulses.